controller/src/main/java/io/pravega/controller/store/stream/tables/State.java

```diff
@@ -9,12 +9,40 @@
  */
 package io.pravega.controller.store.stream.tables;
 
+import com.google.common.collect.Sets;
+
+import java.util.Arrays;
+import java.util.Set;
+
 /**
  * This is used to represent the state of the Stream.
  */
 public enum State {
     UNKNOWN,
     CREATING,
     ACTIVE,
-    SEALED
+    UPDATING,
+    SCALING,
+    SEALING,
+    SEALED;
+
+    private enum StateTransitions {
+        UNKNOWN(State.UNKNOWN, State.CREATING),
+        CREATING(State.CREATING, State.ACTIVE),
+        ACTIVE(State.ACTIVE, State.SCALING, State.SEALING, State.SEALED, State.UPDATING),
+        SCALING(State.SCALING, State.ACTIVE),
+        UPDATING(State.UPDATING, State.ACTIVE),
+        SEALING(State.SCALING, State.SEALED),
+        SEALED(State.SEALED);
+
+        private final Set<State> transitions;
+
+        StateTransitions(State... states) {
+            this.transitions = Sets.immutableEnumSet(Arrays.asList(states));
```

Comment:
Sets.immutableEnumSet has two signatures Sets.immutableEnumSet(E, E...) and Sets.immutableEnumSet(Iterable<E>)
and State... (== State[]) does not satisfy either. Hence Arrays.asList